src/VecSim/algorithms/brute_force/bfm_batch_iterator.h

```diff
@@ -19,34 +19,30 @@ class BFM_BatchIterator : public BF_BatchIterator<DataType, DistType> {
     ~BFM_BatchIterator() override = default;
 
 private:
-    inline VecSimQueryReply_Code calculateScores() override {
+    VecSimQueryReply_Code calculateScores() override {
         this->index_label_count = this->index->indexLabelCount();
         this->scores.reserve(this->index_label_count);
         vecsim_stl::unordered_map<labelType, DistType> tmp_scores(this->index_label_count,
                                                                   this->allocator);
-        VecSimQueryReply_Code rc;
 
         idType curr_id = 0;
-        while (curr_id < this->index->indexSize()) {
-            idType last_id =
-                MIN(curr_id + this->index->getBlockSize() - 1, this->index->indexSize() - 1);
-            // compute the scores for the vectors in every block and extend the scores array.
-            auto block_scores = this->index->computeBlockScores(
-                curr_id, last_id, this->getQueryBlob(), this->getTimeoutCtx(), &rc);
-            if (VecSim_OK != rc) {
-                return rc;
+        auto *vectors_it = this->index->getVectorsIterator();
+        auto DistFunc = this->index->getDistFunc();
+        while (vectors_it->hasNext()) {
+            // Compute the scores for every vector and extend the scores array.
+            if (VECSIM_TIMEOUT(this->getTimeoutCtx())) {
+                return VecSim_QueryReply_TimedOut;
```

Comment:
this is probably OK, but we initially did not check for timeout after every vector to avoid the overhead (only per block). We should at least benchmark to see we don't add a regression here

Reply:
ARe you sure? We had the timeout check in the compute block score function before, in every iteration:

```    
for (size_t i = 0; i < len; i++) {
        if (VECSIM_TIMEOUT(timeoutCtx)) {
            *rc = VecSim_QueryReply_TimedOut;
            return scores;
        }
         ...
}
```

Reply:
OK maybe here we did. But for pure KNN queries we didn't